index.js

```diff
@@ -532,9 +558,17 @@
 
   //  Number$prototype$equals :: Number ~> Number -> Boolean
   function Number$prototype$equals(other) {
-    return typeof this === 'object' ?
-      equals(this.valueOf(), other.valueOf()) :
-      isNaN(this) && isNaN(other) || this === other && 1 / this === 1 / other;
+    return typeof other === 'object' ?
+      typeof this === 'object' &&
+        equals(this.valueOf(), other.valueOf()) :
+      isNaN(other) && isNaN(this) || this === other;
+  }
+
+  //  Number$prototype$lte :: Number ~> Number -> Boolean
+  function Number$prototype$lte(other) {
+    return isNaN(this) ? isNaN(other) :
+      (typeof this === 'object' ? this.valueOf() : this)
+        <= (typeof other === 'object' ? other.valueOf() : other);
   }
```

Comment:
Cool, will do. Are we therefore happy with this behaviour: https://github.com/sanctuary-js/sanctuary-type-classes/pull/45#discussion_r110637365 ?